mreg_cli/utilities/api.py

```diff
@@ -503,25 +503,15 @@ def _check_expect_one_result(
     if limit and resp.count > abs(limit):
         raise TooManyResults(f"Too many hits ({resp.count}), please refine your search criteria.")
 
-    # Short circuit if there are no more pages. This means that there are no more results to
-    # be had so we can return the results we already have.
-    if not resp.next:
-        return _check_expect_one_result(resp.results)
-
     # Iterate over all pages and collect the results
-    ret: list[Json] = []
-    while True:
-        resp = get(path, params=params, ok404=ok404)
-        if resp is None:
-            return _check_expect_one_result(ret)
-        result = validate_paginated_response(resp)
-
-        ret.extend(result.results)
-
-        if result.next:
-            path = result.next
-        else:
+    ret: list[Json] = resp.results
```

Comment:
I considered doing a copy here to avoid mutation of the results, but it really has no value. We are returning one large list anyway, so whether we modify the original or a copy doesn't matter to the caller of the function.

Reply:
Makes sense. I think? :)

Reply:
Yeah... It seems wrong, but _always_ copying the list when we strictly don't need to is sub-optimal. Then again, computers are fast and network requests are not, so maybe this is a silly concern?